indigo_app/templates/indigo_api/task_form.html

```diff
@@ -118,14 +118,26 @@
         </div>
 
         <div class="card-footer text-right">
-          {% if task.state == 'open' or task.state == 'pending_review' %}
+          {% if task.state == 'open' or task.state == 'pending_review' or task.state == 'blocked' %}
             <form method="POST">
             {% csrf_token %}
               {% if cancel_task_permission %}
                 <button class="btn btn-danger float-left" type="submit" data-confirm="Are you sure you want to cancel this task?" formaction="{% url 'cancel_task' place=place.place_code pk=task.pk %}">Cancel task</button>
               {% endif %}
             </form>
           {% endif %}
+          {% if task.state == 'open' and block_task_permission %}
+            <form method="POST">
```

Comment:
Rather than duplicating the form, just wrap them all in the same form, since the `formaction` attribute does the necessary.